etcd3/client.py

```diff
@@ -88,6 +69,68 @@ def __init__(self, alarm_type, member_id):
         self.member_id = member_id
 
 
+class Endpoint(object):
+    """Represents an etcd cluster endpoint.
+
+    :param str host: Endpoint host
+    :param int port: Endpoint port
+    :param bool secure: Use secure channel, default True
+    :param creds: Credentials to use for secure channel, required if
+                  secure=True
+    :type creds: grpc.ChannelCredentials, optional
+    :param time_retry: Seconds to wait before retrying this endpoint after
+                       failure, default 300.0
+    :type time_retry: int or float
+    :param opts: Additional gRPC options
+    :type opts: dict, optional
+    """
+
+    def __init__(self, host, port, secure=True, creds=None, time_retry=300.0,
```

Comment:
300 seconds seems pretty long as a default, is there a particular reason for this length?

Reply:
I just kept what the previous author had used here. Since there's currently no built-in provision for jitter or exponential backoff, setting this to be a bit long may help mitigate any "thundering herds" a little. In the event that the user has set up multi-endpoint failover and there are still health nodes available, their application won't remain unavailable for those five minutes.

I'm not arguing for any particular value (I also think it could be a bit shorter), but figured  I should present an argument for keeping it long.